File: ios/Classes/Bridge/GetDpsMetadataBridge.swift
import Usercentrics

struct GetDpsMetadataBridge : MethodBridge {

    let name: String = "getDpsMetadata"
    let usercentrics: UsercentricsProxyProtocol

    func invoke(_ call: FlutterMethodCall, _ result: @escaping FlutterResult) {
        assert(call.method == name)
        guard let templateId = call.arguments as? String else {
            result(FlutterError(code: "usercentrics_flutter_getDpsMetadata_error",
                                message: "Invalid Parameter",
                                details: nil ))
            return
        }

        result(usercentrics.shared.getDpsMetadata(templateId: templateId))
    }
}

File: android/src/main/kotlin/com/usercentrics/sdk/flutter/bridge/GetDpsMetadataBridge.kt
package com.usercentrics.sdk.flutter.bridge

import com.usercentrics.sdk.flutter.api.FlutterMethodCall
import com.usercentrics.sdk.flutter.api.FlutterResult
import com.usercentrics.sdk.flutter.api.UsercentricsProxy
import com.usercentrics.sdk.flutter.api.UsercentricsProxySingleton

internal class GetDpsMetadataBridge(
    private val usercentrics: UsercentricsProxy = UsercentricsProxySingleton
) : MethodBridge {

    override val name: String
        get() = "getDpsMetadata"

    override fun invoke(call: FlutterMethodCall, result: FlutterResult) {
        assert(name == call.method)
        val templateId = call.arguments as String
        result.success(usercentrics.instance.getDpsMetadata(templateId))
    }
}
